### ISSUE TITLE
Completed Workflows Connected to Connection Request Deletion Issues

### ISSUE THREAD

**Issue by dataCollegechurch** [SparkDevNetwork/Rock#6144]
### Description

Completed workflows Connected launched from a connection request present two issues.
1) They are not automatically cleaned up by the Rock cleanup job after its days completed exceeds the **_Completed Workflow Retention Period (days)_**
2) When you attempt to manually delete them, you are presented with a warning "This Workflow is assigned to a Connection Request Workflow."

### Actual Behavior

Completed workflows Connected launched from a connection request present two issues.
1) They are not automatically cleaned up by the Rock cleanup job after its days completed exceeds the **_Completed Workflow Retention Period (days)_**
![image](https://github.com/user-attachments/assets/ebc3c244-2a72-45ee-af1f-024a2e713486)
![image](https://github.com/user-attachments/assets/b84676de-4f2e-4603-8935-9356a0fcbceb)

2) When you attempt to manually delete them, you are presented with a warning "This Workflow is assigned to a Connection Request Workflow."
![image](https://github.com/user-attachments/assets/c253730f-2abb-4a24-9b44-1cdf90910e5d)

### Expected Behavior

A completed workflow regardless of what it is connected to would be deleted after its days completed exceeds the **_Completed Workflow Retention Period (days)_**. A user can delete a completed workflow that is assigned to a Connection Request Workflow.

### Steps to Reproduce

1) Upload this workflow
[Notify Connector of Future Follow-Up Date Reached_202412300943.json](https://github.com/user-attachments/files/18276337/Notify.Connector.of.Future.Follow-Up.Date.Reached_202412300943.json)
2) Setup the workflow to launch when a future follow-up date is reached
![image](https://github.com/user-attachments/assets/51862153-291c-4aeb-9eeb-6d718041b51a)
3) Set the connector and future follow-up date to today on an active connection request
![image](https://github.com/user-attachments/assets/760b47eb-d121-455a-b9fb-e627fd1ccd8d)
4) Run the Connection Request Workflow Triggers, this will generate a workflow tied to a connection request
![image](https://github.com/user-attachments/assets/042a989a-6f3b-4689-884b-145198cba85e)
5) Attempt to delete completed workflow
6) After the days completed exceeds the **_Completed Workflow Retention Period (days)_**, run the rock clean up job
![image](https://github.com/user-attachments/assets/0156dad3-d8c2-4a3e-88bd-c1973bca0411)
7) See that the completed workflow remains
![image](https://github.com/user-attachments/assets/2f1b09b8-b6c4-4297-9842-28b4f397a764)

### Issue Confirmation

- [X] Perform a search on the [Github Issues](https://github.com/issues?q=is%3Aissue+user%3ASparkDevNetwork+-repo%3ARock) to see if your bug or enhancement is already reported.
- [X] Reproduced the problem on a fresh install or on the [demo site](https://rock.rocksolidchurchdemo.com/).

### Rock Version

16.7

### Client Culture Setting

en-US

**Comment by JimMichael** [comment]
I agree that this is odd behavior, particularly because there is a "Delete" button at the bottom of the Workflow Instance grid
![image](https://github.com/user-attachments/assets/17a4c1b9-5bb7-410d-a55d-3eb65d92130a)
that DOES delete these Connection-Request-Initiated workflows when they are selected and the button is pressed:
![image](https://github.com/user-attachments/assets/687f8b65-2214-4b20-a39f-3201105dfec2)

This implies that it's possible/ok to delete such workflows, but the other deletion methods (manully clicking x and cleanup job) just don't work the same way.

**Comment by dataCollegechurch** [comment]
I am not sure what the label "x-Fixed in v17.0" means. This does not appear to be fixed as of 18.4.1.
